tests/ssl-opt.sh

```diff
@@ -11297,16 +11295,13 @@ requires_config_enabled MBEDTLS_SSL_CLI_C
 run_test    "TLS 1.3: Server side check - mbedtls with client authentication" \
```

Comment:
Please add two tests similar to this one where the client responds to the CertificateRequest message with an empty Certificate message. In the first one `auth_mode=required` on the server and the handshake fails. On the second one `auth_mode=optional` and the handshake completes successfully.